enhancements/installer/vsphere-ipi-zonal.md

```diff
@@ -0,0 +1,1067 @@
+	// This name will be used in the install MachinePool
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// server is the fully-qualified domain name or the IP address of the vCenter server.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server,omitempty"`
+
+	// failureDomain is the name of FailureDomain used for this DeploymentZone
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	FailureDomain string `json:"failureDomain,omitempty"`
+
+	// ControlPlane determines if this failure domain is suitable for use by control plane machines.
+	// +kubebuilder:validation:Optional
+	ControlPlane DeploymentSuitable `json:"controlPlane,omitempty"`
+
+	// PlacementConstraint encapsulates the placement constraints
+	// used within this deployment zone.
+	// +kubebuilder:validation:Required
+	PlacementConstraint PlacementConstraint `json:"placementConstraint"`
+}
+```
+
+Add [platform validation](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/platform.go)
+for the new struct fields that are required.
+
+
+#### Set infrastructure spec
+
+With the changes to the openshift/api infrastructure config
+the [infrastructure manifest asset generation](https://github.com/openshift/installer/blob/master/pkg/asset/manifests/infrastructure.go#L188-L195) will need to be modified to
+generate VSpherePlatformSpec and provide the values from the install-config platform spec.
+
+#### MachinePool
+
+The `MachinePool` [struct](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/machinepool.go#L5-L26)
+needs a single change to include the zones.
+The `Zones` like the other cloud providers will determine the location
+where the virtual machine will reside within a vSphere environment.
+
+```golang
+type MachinePool struct {
+...
+  DeploymentZones []string `json:"deploymentZones,omitempty"`
+}
+```
+
+Add [machinepool validation](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/machinepool.go)
+
+#### Cloud Config
+
+The out-of-tree CCM is required for this change. The out-of-tree CCM also updates
+the cloud-config ini configuration. Below is an example of what the
+cloud-config will change to. A `VirtualCenter` section will be added per
+vcenter in `vcenters`. `datacenters` is a comma-delimited string that will contain
+all the datacenters per region.
+
+```ini
+[Global]
+  insecure-flag = "1"
+  secret-name = "vsphere-creds"
+  secret-namespace = "kube-system"
+
+[VirtualCenter "10.0.0.1"]
+  datacenters = "SDDC-Datacenter"
+
+[Labels]
+  region = openshift-region
+  zone = openshift-zone
+```
+
+There is an outstanding question if the out-of-tree cloud provider will be
+enabled by default. If this does not happen of 4.12 we will need to include
```

Comment:
This sentence needs double checking, `does not happen of`?

Reply:
ack